docs/oauth/oauth-scoped-access.mdx

```diff
@@ -0,0 +1,125 @@
+
+  ### Test the OAuth flow
+
+  In order to test the OAuth flow, you can use the following [OAuth test client](https://github.com/clerk/test-oauth-client) project. It contains a minimal Express.js server that can run through an OAuth flow.
+
+  > [!NOTE]
+  > This implementation follows [the OAuth specification](https://datatracker.ietf.org/doc/html/draft-ietf-oauth-v2-1-13), and is not specific to Clerk or this test project. Any compliant OAuth client will behave the same way.
+
+  1. Clone the [following repository](https://github.com/clerk/test-oauth-client) to your machine.
+
+  1. In your terminal and within your project, run the following command:
+
+     ```bash {{ filename: 'terminal' }}
+     cp .env.sample .env
+     ```
+
+  1. Add the required credentials to this newly created `.env` file:
+
+     - `CLIENT_ID`: The client ID of your OAuth app from the Clerk Dashboard.
+     - `CLIENT_SECRET`: The client secret of your OAuth app from the Clerk Dashboard.
+     - `NEXT_PUBLIC_CLERK_PUBLISHABLE_KEY`: Your Clerk Publishable Key from the [**API keys**](https://dashboard.clerk.com/last-active?path=api-keys) page in the Clerk Dashboard.
+
+  1. In your terminal, run the following command to install dependencies:
+
+     ```bash {{ filename: 'terminal' }}
+     npm i
+     ```
+
+  1. In your terminal, run the following command to start the server:
+
+     ```bash {{ filename: 'terminal' }}
+     npm start
+     ```
+
+  ### Finished 🎉
+
+  Once this is complete, you will be able to see the OAuth flow begin. If you are not currently signed in, you will first be presented with a sign in screen. After signing in, Clerk will redirect to an OAuth consent screen, showing the requested scopes and asking for your consent.
+
+  > [!NOTE]
+  > The consent screen uses the scopes passed in the authorization request to inform the user exactly what they’re granting access to, and to whom.
+  >
+  > By default, the consent screen is shown for all newly created OAuth apps, but this can be toggled in each app’s settings [on the Clerk Dashboard](https://dashboard.clerk.com/last-active?path=oauth-applications), although not recommended.
+  >
+  > For more details on how our Clerk OAuth consent screen works, see the [OAuth consent screen reference docs](/docs/oauth/how-clerk-implements-oauth#consent-screen-management).
+
+  Once you have accepted, you will be redirected to the OAuth callback route, the `redirect_uri` specified earlier. This process will exchange the authorization code for an access token, and render the access token and refresh token as a JSON response, similar to this:
+
+  ```json
+  {
+    "access_token": "xxx",
+    "expires_in": 7200,
+    "id_token": "xxx",
+    "refresh_token": "xxx",
+    "scope": "openid email profile",
+    "token_type": "bearer"
+  }
+  ```
+</Steps>
+
+## Accepting and verifying an OAuth Token with Clerk SDKs
```

Comment:
I do still think we should bounce this out to a separate page, then this section at the bottom can link out to that page with something like "with the OAuth flow complete, the client now has a working access token. The next step would normally be for the client to use that token to make an authenticated request to your app's API endpoints. To learn more about how to accept and verify OAuth access tokens using Clerk, see our guide on verifying oauth access tokens (link)"